scripts/testing-utils.js

```diff
@@ -115,8 +119,171 @@ function launchPackagerInSeparateWindow(folderPath) {
+      throw new Error(
+        `The ${workflow.name} workflow status is ${workflow.status}. Please, wait for it to be finished before start testing or fix it`,
+      );
+    }
+  }
+
+  async #getLastCircleCIPipelineID(branchName) {
+    const options = {
+      method: 'GET',
+      url: 'https://circleci.com/api/v2/project/gh/facebook/react-native/pipeline',
+      qs: {
+        branch: branchName,
+      },
+      headers: this.circleCIHeaders,
+    };
+
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const lastPipeline = JSON.parse(response.body).items[0];
+    return {id: lastPipeline.id, number: lastPipeline.number};
+  }
+
+  async #getSpecificWorkflow(pipelineId, workflowName) {
+    const options = {
+      method: 'GET',
+      url: `https://circleci.com/api/v2/pipeline/${pipelineId}/workflow`,
+      headers: this.circleCIHeaders,
+    };
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const body = JSON.parse(response.body);
+    return body.items.find(workflow => workflow.name === workflowName);
+  }
+
+  async #getPackageAndReleaseWorkflow(pipelineId) {
+    return this.#getSpecificWorkflow(
+      pipelineId,
+      'package_and_publish_release_dryrun',
+    );
+  }
+
+  async #getTestsWorkflow(pipelineId) {
+    return this.#getSpecificWorkflow(pipelineId, 'tests');
+  }
+
+  async #getCircleCIJobs(workflowId) {
+    const options = {
+      method: 'GET',
+      url: `https://circleci.com/api/v2/workflow/${workflowId}/job`,
+      headers: this.circleCIHeaders,
+    };
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const body = JSON.parse(response.body);
+    return body.items;
+  }
+
+  async #getJobsArtifacts(jobNumber) {
+    const options = {
+      method: 'GET',
+      url: `https://circleci.com/api/v2/project/gh/facebook/react-native/${jobNumber}/artifacts`,
+      headers: this.circleCIHeaders,
+    };
+    const response = await request(options);
+    if (response.error) {
+      throw new Error(error);
+    }
+
+    const body = JSON.parse(response.body);
+    return body.items;
+  }
+
+  async #findUrlForJob(jobName, artifactPath) {
+    const job = this.jobs.find(j => j.name === jobName);
+    const artifacts = await this.#getJobsArtifacts(job.job_number);
+    return artifacts.find(artifact => artifact.path.indexOf(artifactPath) > -1)
+      .url;
+  }
+
+  async artifactURLHermesDebug() {
+    return this.#findUrlForJob(
+      'build_hermes_macos-Debug',
+      'hermes-ios-debug.tar.gz',
+    );
+  }
+
+  async artifactURLForMavenLocal() {
+    return this.#findUrlForJob(
+      'build_and_publish_npm_package-2',
+      'maven-local.zip',
+    );
+  }
+
+  async artifactURLForPackagedReactNative() {
+    return this.#findUrlForJob(
+      'build_and_publish_npm_package-2',
+      'react-native-1000.0.0-',
```

Comment:
are we sure we always need the `-1000.0.0` version? will this script work on release branches?

You can test that by rebasing your code locally on top of ex. `0.72-stable` to verify that that's the case.

Reply:
I looked at the RC versions of 0.72 to get these values.
[These are the artifacts](https://app.circleci.com/pipelines/github/facebook/react-native/26364/workflows/f315f1fb-3335-494a-967d-27db83f6f3dc/jobs/770169/artifacts) of the last job that run on 0.72-stable. And React Native has been packaged with `1000.0.0` as a version.

The [publish-release job](https://app.circleci.com/pipelines/github/facebook/react-native/26243/workflows/18885b2d-2a00-416d-9eab-3fb012be4e09/jobs/763359/artifacts) which publishes stuff to npm does not even upload the artifacts to be downloaded